r/vignettes/developers/install_details.Rmd

```diff
@@ -0,0 +1,123 @@
+# How the R package is installed
+
+In order for the arrow R package to work, it needs the Arrow C++ library, 
+also known as libarrow.  There are a number of scripts that are triggered 
+when `R CMD INSTALL .` is run and for Arrow users, these should all just work 
+without configuration and pull in the most complete pieces (e.g. official 
+binaries that we host).
+
+This documented is intended specifically for arrow _developers_ who wish to know 
+more about these scripts.  If you are an arrow _user_ looking for help with 
+installing arrow, please see [the installation guide](../install.html)
+
+An overview of these scripts is shown below:
+
+* `configure` and `configure.win` - these scripts are triggered during
+`R CMD INSTALL .` on non-Windows and Windows platforms, respectively. They
+handle finding the libarrow, setting up the build variables necessary, and
+writing the package Makevars file that is used to compile the C++ code in the R
+package.
+
+* `tools/nixlibs.R` - this script is sometimes called by `configure` on Linux
+(or on any non-windows OS with the environment variable
+`FORCE_BUNDLED_BUILD=true`) if an existing libarrow installation cannot be found.
+This sets up the build process for our bundled builds (which is the default on 
+linux) and checks for binaries or downloads libarrow from source depending on 
+dependency availability and build configuration.
+
+* `tools/winlibs.R` - this script is sometimes called by `configure.win` on Windows
+when environment variable `ARROW_HOME=true`.  It looks for an existing libarrow
+installation, and if it can't find one downloads an appropriate libarrow binary.
+
+* `inst/build_arrow_static.sh` - called by `tools/nixlibs.R` when libarrow  
+needs to be built.  It builds libarrow for a bundled, static build, and
+mirrors the steps described in the ["Arrow R Developer Guide" vignette](./setup.html)
+This build script is also what is used to generate our prebuilt binaries.
+
+The actions taken by these scripts to resolve dependencies and install the 
+correct components are described below.
+
+There are a number of different ways you can install libarrow:
```

Comment:
These two paragraphs seem a bit disjointed? Should there be a `### libarrow` line between them?  Or some other kind of introduction to talking about libarrow (specifically)?

Reply:
Yeah, I agree - have rearranged content now to hopefully flow better